File: Hello_Arg/Hello_Arg.c

#include <stdio.h>

void showArgs(int argc, char **argv);

int main(int argc, char **argv) {
    printf("Hello World from C\n");
    showArgs(argc, argv);

    return 0;
}

void showArgs(int argc, char **argv) {

    int i;

    int args=argc-1;

    printf("You provided %d args to command %s\n", args, argv[0]);
    for(i=1; i<argc; i++) {
        printf("arg[%d]='%s'\n", i, argv[i]);
    }
}

File: Hello_Arg/Hello_Arg.pl
#!/usr/bin/perl -w

use strict;

# Of course there's(way) more than one way to do it [(W)MTOWTDI] in Perl!
print "Hello World from Perl!!\n";
showArgs(\@ARGV);

sub showArgs {
    my $argsRef = shift;

    my $i=0;

    printf "You provided %d args to command %s\n", scalar(@$argsRef), $0;
    while ($_ = shift(@$argsRef)) {
        $i++;
        print "arg[$i]='$_'\n";
    }
}

